manifests/init.pp

```diff
@@ -56,13 +77,30 @@
   $manage_user          = false,
   $user                 = 'root',
   $group                = undef,
-  $manage_repo          = false
+  $manage_repo          = false,
+  $authenticate         = false,
+  $secret_file          = '/etc/mesos/marathon.secret',
+  $auth_principal       = undef,
+  $auth_secret          = undef,
+  $auth_role            = undef
+
 ) inherits marathon::params {
 
   validate_bool($install_java)
   validate_bool($service_enable)
   validate_bool($manage_user)
   validate_bool($manage_repo)
+  validate_bool($authenticate)
+
+  if $authenticate {
+    if $auth_principal == undef {
+      fail('You must provide a principal when using authentication.')
+    }
+    if $auth_secret == undef {
+      fail('You must provide a secret when using authentication.')
```

Comment:
Would be nice if this could be left unspecified, so that it's easier to manage the secret file out of band (e.g. with vault / consul-template)